packages/fx-core/src/component/generator/generatorAction.ts

```diff
@@ -174,33 +202,12 @@ export const fetchTemplateFromLocalAction: GeneratorAction = {
   },
 };
 
-export const unzipAction: GeneratorAction = {
-  name: GeneratorActionName.Unzip,
-  run: async (context: GeneratorContext) => {
-    if (!context.zip) {
-      return;
-    }
-    context.logProvider.debug(`Unzipping: ${JSON.stringify(context)}`);
-    context.outputs = await unzip(
-      context.zip,
-      context.destination,
-      context.fileNameReplaceFn,
-      context.fileDataReplaceFn,
-      context.filterFn
-    );
-  },
-};
-
 export const TemplateActionSeq: GeneratorAction[] = [
   fetchTemplateZipFromSourceCodeAction,
-  fetchTemplateUrlWithTagAction,
+  fetchUrlForHotfixOnlyAction,
   fetchZipFromUrlAction,
   unzipAction,
   fetchTemplateFromLocalAction,
```

Comment:
I think the new flow could be:
1. Set templates zip in context if debug flag enabled (We will deprecate this step later).
2. (Skip if zip set or useLocal flag is true) Get latest templates version from GitHub tag list and set tag in context if there is a hotfix.
3. (Skip if zip set or tag not set or useLocal flag is true) Download templates zip from GitHub tag and set zip.
4. (Skip if zip not set) unzip, set outputs.
5. (Skip if outputs set) unzip local templates zip.

You can make more changes in the source code if necessary.

Reply:
To simplify the flow, maybe you can consider merging `fetchUrlForHotfixOnlyAction`, `fetchZipFromUrlAction`, `unzipAction` into one action.

Reply:
Agree to do some refactor to simplify Generator as well.